File: python/solutions/solution2.py


def fibbonaci_solver(a, b, limit, list):
    c = a + b
    if c > limit:
        return
    if c % 2 == 0:
        list.append(c)
    fibbonaci_solver(b, c, limit, list)


def solution(limit):
    even_numbers = []
    fibbonaci_solver(1, 1, limit, even_numbers)
    return sum(even_numbers)


if __name__ == "__main__":
    print(f"The solution is {solution(4E6)}")

File: haskell/solutions/solution2.hs
module Solution2 (
    solve2
) where


fibs_limited :: Int -> [Int]
fibs_limited limit =
   let fibs = 0 : 1 : [ n | x <-[2..], let n = ((fibs !! (x-1)) + (fibs !! (x-2)))]
   in takeWhile (<limit) fibs

solve2 :: Int -> Int
solve2 limit = sum(filter even (fibs_limited limit))
